### PR TITLE
Implement organization level filtering for scanning resources

### LINKED ISSUE #343: Implement organization level filter for resources
ΩFidesctl being extended to support scanning of infrastructure components could make it possible to annotate where organizations store their data. Knowing where pii data is stored is important for allowing users to meet RoPA requirements. 

As part of #340 and #341 the `generate system` and `scan system` commands should exist for aws accounts. This issue extends this feature to support filtering of resources. One use case for tihs feature could be to ignore resources which hold test data. 

Requirements:
* Define a filter model at the organization level for filtering aws resources. Investigate potentiallky using AWS query filters so we don't invent our own pattern if not needed.
* Use model to filter resources

Sample Model:
```
organization:
  fides_key: default_organization
  resource_filters: [
    {filter_type: aws_resource, filter: "<AWS ARN>"}
  ]
```

### PR DESCRIPTION
**Pull request by earmenda**
Closes #343

### Code Changes

* [x] Organization is now an optional input to `scan system aws` and `generate system aws` commands
* [x] Organization now has a `fidesctl_meta` field
* [x] Organization `fidesctl_meta` contains a ResourceFilter list. ResourceFilter contains just a filter type and value. 
* [x] Organization is now fetched for `scan system aws` and `generate system aws` commands
* [x] Created new `filters.py` module to filter systems. Wrote it in a way where adding new filter types would be simple. 
* [x] Updated `generate_system_aws` and `scan_system_aws` functions to filter systems calling `filter_aws_systems`

### Steps to Confirm

* [x] Unit tests for new logic
* [x] Manual testing confirms filtering works as expected

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded

### Description Of Changes

* Biggest caveat with this change is the generate and scan commands now working with an organization. I think this is a positive change as `default_organization` was being assumed in some places already. 

#### No Filtering

AWS Account Contains:
```
arn:aws:redshift:us-east-1:910934740016:namespace:a3f04b4e-b044-4ba5-9aa9-57cddac54305
arn:aws:redshift:us-east-1:910934740016:namespace:057d5b0e-7eaa-4012-909c-3957c7149176
```

Manifest Contains:
```
system:
- fides_key: redshift-cluster-1
  organization_fides_key: default_organization
  name: redshift-cluster-1
  description: 'Fides Generated Description for Redshift Cluster: redshift-cluster-1'
  fidesctl_meta:
    resource_id: arn:aws:redshift:us-east-1:910934740016:namespace:057d5b0e-7eaa-4012-909c-3957c7149176
    endpoint_address: redshift-cluster-1.c2angfh5kpo4.us-east-1.redshift.amazonaws.com
    endpoint_port: '5439'
  system_type: redshift_cluster
  privacy_declarations: []
  administrating_department: Not defined
```

Scan Output
```
Scanned 2 resource(s) and found 1 missing resource(s).
Missing Redshift Cluster:
	arn:aws:redshift:us-east-1:910934740016:namespace:a3f04b4e-b044-4ba5-9aa9-57cddac54305

Resource coverage: 50%
```

#### Exact Filtering of Missing Resource

AWS Account Contains:
```
arn:aws:redshift:us-east-1:910934740016:namespace:a3f04b4e-b044-4ba5-9aa9-57cddac54305
arn:aws:redshift:us-east-1:910934740016:namespace:057d5b0e-7eaa-4012-909c-3957c7149176
```

Manifest Contains:
```
system:
- fides_key: redshift-cluster-1
  organization_fides_key: default_organization
  name: redshift-cluster-1
  description: 'Fides Generated Description for Redshift Cluster: redshift-cluster-1'
  fidesctl_meta:
    resource_id: arn:aws:redshift:us-east-1:910934740016:namespace:057d5b0e-7eaa-4012-909c-3957c7149176
    endpoint_address: redshift-cluster-1.c2angfh5kpo4.us-east-1.redshift.amazonaws.com
    endpoint_port: '5439'
  system_type: redshift_cluster
  privacy_declarations: []
  administrating_department: Not defined
organization:
- fides_key: default_organization
  name: default_organization
  fidesctl_meta:
    resource_filters: [
      {type: "ignore_resource_arn", value: "arn:aws:redshift:us-east-1:910934740016:namespace:a3f04b4e-b044-4ba5-9aa9-57cddac54305"},    ]
```

Scan Output
```
Scanned 1 resources and all were found in taxonomy.
Resource coverage: 100%
```

#### Wildcard Filtering of All Redshift Clusters

AWS Account Contains:
```
arn:aws:redshift:us-east-1:910934740016:namespace:a3f04b4e-b044-4ba5-9aa9-57cddac54305
arn:aws:redshift:us-east-1:910934740016:namespace:057d5b0e-7eaa-4012-909c-3957c7149176
```

Manifest Contains:
```
system:
- fides_key: redshift-cluster-1
  organization_fides_key: default_organization
  name: redshift-cluster-1
  description: 'Fides Generated Description for Redshift Cluster: redshift-cluster-1'
  fidesctl_meta:
    resource_id: arn:aws:redshift:us-east-1:910934740016:namespace:057d5b0e-7eaa-4012-909c-3957c7149176
    endpoint_address: redshift-cluster-1.c2angfh5kpo4.us-east-1.redshift.amazonaws.com
    endpoint_port: '5439'
  system_type: redshift_cluster
  privacy_declarations: []
  administrating_department: Not defined
organization:
- fides_key: default_organization
  name: default_organization
  fidesctl_meta:
    resource_filters: [
      {type: "ignore_resource_arn", value: "arn:aws:redshift:us-east-1:::"},    ]
```

Scan Output
```
Did not find any resources to scan in AWS account
```

### CODE REVIEW

**Comment by earmenda** [comment]
@ThomasLaPiana @NevilleS  Can i get your thoughts on the filtering implementation here? 

I put some examples in the main body of the PR. 

The way a filter is defined is within a fidesctl_meta within an organization:
```
organization:
- fides_key: default_organization
  name: default_organization
  fidesctl_meta:
    resource_filters: [
      {type: "ignore_resource_arn", value: "arn:aws:redshift:us-east-1:::"},    
    ]
```

The filter can exclude any resources with an exact matching arn and also supports wildcarding individual arn fields. An empty arn field in the filter pattern works as a wildcard, following the same pattern as IAM . 

Filter value to exclude an exact arn for a redshift cluster - `arn:aws:redshift:us-east-1:910934740016:namespace:057d5b0e-7eaa-4012-909c-3957c7149176`
Filter value to exclude all redshift clusters in us-east-1 for any account - `arn:aws:redshift:us-east-1:::`

File: fidesctl/src/fideslang/models.py [hunk 1]
```diff
@@ -322,7 +342,7 @@ class SystemMetadata(BaseModel):
     """
     The SystemMetadata resource model.
 
-    asdasdasas
```
Comment (ThomasLaPiana) [inline fidesctl/src/fideslang/models.py:325]: 😆

**Comment by ThomasLaPiana** [comment]
> @ThomasLaPiana @NevilleS Can i get your thoughts on the filtering implementation here?
> 
> I put some examples in the main body of the PR.
> 
> The way a filter is defined is within a fidesctl_meta within an organization:
> 
> ```
> organization:
> - fides_key: default_organization
>   name: default_organization
>   fidesctl_meta:
>     resource_filters: [
>       {type: "ignore_resource_arn", value: "arn:aws:redshift:us-east-1:::"},    
>     ]
> ```
> 
> The filter can exclude any resources with an exact matching arn and also supports wildcarding individual arn fields. An empty arn field in the filter pattern works as a wildcard, following the same pattern as IAM .
> 
> Filter value to exclude an exact arn for a redshift cluster - `arn:aws:redshift:us-east-1:910934740016:namespace:057d5b0e-7eaa-4012-909c-3957c7149176` Filter value to exclude all redshift clusters in us-east-1 for any account - `arn:aws:redshift:us-east-1:::`

this makes sense to me from a `sanity check` standpoint, can't think of an obvious way to make this better. I like having the filter at the org level

File: fidesctl/src/fidesctl/cli/generate_commands.py [hunk 2]
```diff
@@ -47,10 +47,12 @@ def generate_system(ctx: click.Context) -> None:
 @click.pass_context
 @click.argument("output_filename", type=str)
 @include_null_flag
+@click.option("-o", "--organization", type=str, default="default_organization")
```
[the remark this answers is not in the document]
Comment (earmenda) [reply fidesctl/src/fidesctl/cli/generate_commands.py:50]: idea by @NevilleS :)

**Comment by earmenda** [comment]
> Looks fantastic! Only nit is to consider adding some subdirectories to `core` since its getting a bit crowded, but definitely not a blocker

thanks, I will keep that in mind as we add more things. I don't have a clear idea right now of how to structure that yet

**Pull request merged**